### ISSUE TITLE
[Windows] failed to deploy workload to the Windows node in airgap setup

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#23000]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Follow the instruction to prepare the private registry: https://github.com/rancher/docs/issues/1742#issuecomment-532553582 
- Follow the Rancher documentation to install Rancher in airgap HA mode
- add a custom cluster with both Linux and Windows node 
- deploy a workload to the windows node using an image from my private registry, 
the image is `ec2-18-144-18-167.us-west-1.compute.amazonaws.com:443/rancher/maiwj/testcontainer:windows-1903`

**Result:**
- the pod in the workload is stuck in `FailedCreatePodSandBox` state
- The pod tries to pull the image `rancher/kubelet-pause:v0.1.3` from DockerHub instead of the private registry
```
Failed create pod sandbox: rpc error: code = Unknown desc = 
failed pulling image "rancher/kubelet-pause:v0.1.3": 
Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: 
request canceled while waiting for connection 
(Client.Timeout exceeded while awaiting headers)
```

<img width="1312" alt="Screen Shot 2019-09-20 at 3 06 05 PM" src="https://user-images.githubusercontent.com/6218999/65363111-15a6cc00-dbbf-11e9-9f8f-55e3eaed64af.png">
 

**Other details that may be helpful:**

- I log in to the Windows node and I can pull the image from the private registry 
 
- I can deploy a workload to the Linux node with no problems, the image is `ec2-18-144-18-167.us-west-1.compute.amazonaws.com:443/rancher/busybox:latest`
- so the private registry is set up properly in the cluster

**Comment by jiaqiluo** [comment]
`docker inspect kubectl` in the Linux worker node
[linux_kubelet.txt](https://github.com/rancher/rancher/files/3638090/linux_kubelet.txt)

`docker inspect kubectl` in the Windows worker node
[win_kubelet.txt](https://github.com/rancher/rancher/files/3638093/win_kubelet.txt)

**Comment by jiaqiluo** [comment]
The bug fix is validated in v2.3.0-rc8

Following the same steps from the issue, now I can deploy the workload to the Windows node successfully. 
Also, cluster monitoring, project monitoring, creating and restoring a snapshot work in the cluster.
